### ISSUE TITLE
Quarterly Date Filter different on Dashboard than on Question

### ISSUE THREAD

**Issue by hundredwatt** [metabase/metabase#11643]
**Describe the bug**
When clicking on the a question from a dashboard with a quarterly date filter, the "<FIELD> is Q4 - 2019" filter generates SQL that causes records with a date equal to the last date in the quarter to be excluded. 

**Update**: We originally submitted this bug report for v0.33.7.3. We then tried to deploy v0.34.0 and the problem got worse. In v0.34.0, the problem is now present on the dashboard AND the question AND further, records from the last day of the previous quarter are included: https://github.com/metabase/metabase/issues/11643#issuecomment-571341318

**To Reproduce**
1. Create a dashboard with a quarterly date filter and add a question using that filter
2. Set the date filter on the dashboard to "Q4 - 2019"
3. Insert a record in the data with a value in the filter field of "Dec 31, 2019 12:00AM"
4. Click on the question from the dashboard
5. See the question 

**Expected behavior**
Expected: Question result includes records with a value of "Dec 31, 2019 12:00AM" in the date field.

**Screenshots**
The question on the dashboard (with filter):
![Question on Dashboard](https://user-images.githubusercontent.com/91577/71852059-e355a780-30a5-11ea-8a1d-5ee43fb7062b.png)

What you see after clicking on the question:
![After Clicking](https://user-images.githubusercontent.com/91577/71852092-f1a3c380-30a5-11ea-8fe4-2ee7a79a38cb.png)

Here's the same question/filters with a table view:
![Table - After Clicking](https://user-images.githubusercontent.com/91577/71852156-1304af80-30a6-11ea-89bb-f4a85a30d350.png)

As a workaround, I changed the filter from "FIELD is Q4 - 2019" to "FIELD between October 1, 2019 and December 31, 2019". The question now shows the same result as the dashboard:
![Change Filter to Between](https://user-images.githubusercontent.com/91577/71852228-40515d80-30a6-11ea-9697-f6be56461ace.png)

 And if I change the visualization to Table, the row with "Dec 31, 2019 12:00AM" in the field I'm filtering on appears:
![Table - Change Filter to Between](https://user-images.githubusercontent.com/91577/71852278-595a0e80-30a6-11ea-923b-a4579cc817ae.png)

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Your browser and the version: Chrome/79.0.3945.88 
- Your operating system: MacOS 10.4.6
- Your databases: Redshift
- Metabase version: v0.33.7.3 => v0.34.0 is also broken, but in a different way. see my comment
- Metabase hosting environment: Elastic Beanstalk
- Metabase internal database: Postgres

**Severity**
Critical, dashboards are showing in correct data in v0.34.0

**Additional context**
The SQL generated at the Question when the filter is "FIELD is Q4 - 2019" is:
![After Clicking - SQL](https://user-images.githubusercontent.com/91577/71852349-84446280-30a6-11ea-84fd-a8621c24ff6b.png)

While the SQL when the filter is "FIELD between October 1, 2019 and December 31, 2019" is:
![Change Filter to Between - SQL](https://user-images.githubusercontent.com/91577/71852344-80184500-30a6-11ea-8f12-506591d65115.png)

The hour is different in the filter timestamp (4 in the first case, 5 in the second). The timezone for our Metabase instance is US/Eastern. Could this bug be caused by some sort of DST related issue?

**Comment by hundredwatt** [comment]
I found a similar issue, but for filtering on a single date: https://github.com/metabase/metabase/issues/9090

**Comment by hundredwatt** [comment]
We tried updating to v0.34.0 and now Dec 31 is not included in the Dashboard or the Question.

Additionally, Sept 30, 2019 is now being included in Q4 on both the Dashboard and the Question:
![image](https://user-images.githubusercontent.com/91577/71853261-cbcbee00-30a8-11ea-9bd6-bb08d9b9a895.png)

In other words: In v0.34.0, the Dashboard and the Question now show the same, incorrect value.

**Comment by mazameli** [comment]
@camsaul, any reason to think this might've gotten fixed by anything we just shipped in 0.34.1?

**Comment by hundredwatt** [comment]
@dacort Sorry for the delay! 

> Your Diagnostic Info from the Admin --> Troubleshooting page
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.3+7",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.3",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.3+7",
    "os.name": "Linux",
    "os.version": "4.14.97-74.72.amzn1.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "redshift",
      "googleanalytics"
    ],
    "hosting-env": "elastic-beanstalk",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "date": "2019-12-16",
      "tag": "v0.33.7.3",
      "branch": "release-0.33.x",
      "hash": "7ee7193"
    },
    "settings": {
      "report-timezone": "US/Eastern"
    }
  }
}
```

> The schema of the table in question, or at least that specific column (I'm mostly curious if it's a TIMESTAMPTZ or TIMESTAMP type column)

It is a DATE column
